### DISCUSSION TITLE
[v8] - How do I setup external ColumnFilter components?

### DISCUSSION THREAD

**Discussion by jasperdunn** [TanStack/table#4361] [Q&A] [answered]
All the documentation and examples I've seen for filtering use state.columns inside the Table component, with `onChange={value => column.setFilterValue(value)}`.

Is there a way I can expose this functionality to the parent component of my Table?

For context, my goal is to have a dashboard with filters at the top of the page that when set, filter down through all the various charts and tables on the screen.

Thanks

**Comment by jasperdunn** [comment, marked as answer]
This discussion can be closed.
Aggregating data like what I was attempting isn't a good approach in the front-end.
We moved to backend filtering and sorting.

**Reply by markusguenther** [reply]
For me, it would be interesting to achieve that. I don't have so much data that filtering on the backend side makes sense. But I also want to have column filters but on the top of the table or better said outside.

Did not see something in the docs or examples.
So if someone has a hint, it would be nice :)

**Comment by markusguenther** [comment]
I found a way with `useState()` and bit of glue code :)

**Reply by jasperdunn** [reply]
Awesome! I’d be keen to see how you did it 👍

**Reply by tacosjs** [reply]
> I found a way with `useState()` and bit of glue code :)

Mind to elaborate?

**Reply by goritm** [reply]
I'm trying to solve exactly the same problem, would you mind sharing how you solved it? 🙏🏼  @markusguenther

**Reply by markusguenther** [reply]
Have not that mich time at the moment. And need to create a example as I could not use project code. But will try to create a sandbox.
